cpp/demo/mixed_poisson/main.cpp

```diff
@@ -211,8 +201,50 @@ int main(int argc, char* argv[])
           return marker;
         });
 
-    // Compute facets with \sigma (flux) boundary condition facets, which is
-    // {all boundary facet} - {u0 boundary facets }
+    // We'd like to represent `u_0` using a function space defined only
+    // on the facets in dfacets. To do so, we begin by calling
+    // `create_submesh` to get a `submesh` of those facets. It also
+    // returns a map `submesh_to_mesh` whose `i`th entry is the facet in
+    // mesh corresponding to cell `i` in submesh.
+    int tdim = mesh->topology()->dim();
+    int fdim = tdim - 1;
+    std::shared_ptr<mesh::Mesh<U>> submesh;
+    std::vector<std::int32_t> submesh_to_mesh;
+    {
+      auto [_submesh, _submesh_to_mesh, v_map, g_map]
+          = mesh::create_submesh(*mesh, fdim, dfacets);
+      submesh = std::make_shared<mesh::Mesh<U>>(std::move(_submesh));
+      submesh_to_mesh = std::move(_submesh_to_mesh);
+    }
+
+    // Create an element for `u_0`. Since the cells in submesh are
+    // intervals, we use the interval cell type.
+    auto Qe = std::make_shared<fem::FiniteElement<U>>(basix::create_element<U>(
+        basix::element::family::P, basix::cell::type::interval, 1,
```

Comment:
Should we use some `entity_type` wrapper here to get the `interval` from the topological dimension, i.e. https://github.com/FEniCS/basix/blob/8e9c3 or https://github.com/FEniCS/basix/blob/8e9c34720f2bf8646cdb6bd61593a48b4dc5c98f/cpp/basix/cell.h#L122 (check for single entity type).